dashboard/app/models/stage.rb

```diff
@@ -250,9 +250,19 @@ def last_progression_script_level
     script_levels.reverse.find(&:valid_progression_level?)
   end
 
-  def next_level_path_for_stage_extras(user)
+  def next_level_for_stage_extras(user)
     level_to_follow = script_levels.last.next_level
     level_to_follow = level_to_follow.next_level while level_to_follow.try(:locked_or_hidden?, user)
-    level_to_follow ? build_script_level_path(level_to_follow) : script_completion_redirect(script)
+    level_to_follow
+  end
+
+  def next_level_path_for_stage_extras(user)
+    next_level_for_stage_extras(user) ?
+      build_script_level_path(next_level_for_stage_extras(user)) : script_completion_redirect(script)
```

Comment:
it looks like `next_level_for_stage_extras` is hitting the database, so it would be nice to do it fewer times like this:
```
next_level = next_level_for_stage_extras(user)
next_level ? ...
```
here and below.